VFP: Create and use file readers with no unit conversion

magnesj opened the pull request:
Closes #13269

kriben commented on ApplicationLibCode/FileInterface/RifVfpInjTable.h:
```diff
@@ -0,0 +1,93 @@
+    std::array<size_t, 2> shape() const;
+    double                operator()( size_t thp_idx, size_t flo_idx ) const;
+
+    static void check_axis( const std::vector<double>& axis );
```
Avoid snake case?

magnesj replied:
fixed

kriben commented on ApplicationLibCode/FileInterface/RifVfpProdTable.h:
```diff
@@ -0,0 +1,128 @@
+public:
+    enum class FLO_TYPE
+    {
+        FLO_OIL = 1,
+        FLO_LIQ,
+        FLO_GAS
+    };
+
+    enum class WFR_TYPE
+    {
+        WFR_WOR = 11,
+        WFR_WCT,
+        WFR_WGR
+    };
+
+    enum class GFR_TYPE
+    {
+        GFR_GOR = 21,
+        GFR_GLR,
+        GFR_OGR
+    };
+
+    enum class ALQ_TYPE
+    {
+        ALQ_GRAT = 31,
+        ALQ_IGLR,
+        ALQ_TGLR,
+        ALQ_PUMP,
+        ALQ_COMP,
+        ALQ_BEAN,
+        ALQ_UNDEF
+    };
```
Maybe comment on where the value come from? A bit magical now.

magnesj replied:
Not sure where the numbering comes from, use same as opm-common.

kriben commented on ApplicationLibCode/FileInterface/RifVfpProdTable.h:
```diff
@@ -0,0 +1,128 @@
+    std::vector<double> m_wfr_data;
+    std::vector<double> m_gfr_data;
+    std::vector<double> m_alq_data;
+    std::vector<double> m_data;
```
Avoid snake case.

magnesj replied:
fixed

kriben commented on ApplicationLibCode/FileInterface/RifVfpInjTable.cpp:
```diff
@@ -0,0 +1,194 @@
+
+    if ( table.size() < 4 )
+    {
+        throw std::invalid_argument( "VFPINJ table does not appear to have enough records to be valid" );
```
I would prefer a static "createFromOpm" method that could return a std::expected instead of the throwing exceptions. But I also see that it tries to mimic the original OPM class, so might be OK.

magnesj replied:
using same concept as opm-common

kriben commented on ApplicationLibCode/FileInterface/RifVfpProdTable.cpp:
```diff
@@ -0,0 +1,267 @@
+    size_t nw = m_wfr_data.size();
+    size_t ng = m_gfr_data.size();
+    size_t na = m_alq_data.size();
+    size_t nf = m_flo_data.size();
```
Avoid creating variable for these (especially when they are this cryptic).

magnesj replied:
keep for compability with opm-common

kriben commented on ApplicationLibCode/FileInterface/RifVfpProdTable.cpp:
```diff
@@ -0,0 +1,267 @@
+    assert( wfr_idx < nw );
+    assert( gfr_idx < ng );
+    assert( alq_idx < na );
+    assert( flo_idx < nf );
```
CAF_ASSERT?

magnesj replied:
fixed

kriben commented on ApplicationLibCode/FileInterface/RifVfpProdTable.cpp:
```diff
@@ -0,0 +1,267 @@
+    assert( wfr_idx < nw );
+    assert( gfr_idx < ng );
+    assert( alq_idx < na );
+    assert( flo_idx < nf );
```
CAF_ASSERT?

magnesj replied:
fixed

magnesj committed:
Add VFP table importers without unit conversion

The opm-common importers does unit conversion to SI. Create custom importers with no unit conversion.

magnesj committed:
#13269 Read VFP data without unit conversion

- read VFP data using readers with no unit conversion
- remove existing unit conversion
- make sure correct units are displayed in UI for both Metric and Field

kriben reviewed (approved):
Better now!
